swifter/swifter.py

```diff
@@ -205,6 +204,15 @@ def resample(
         }
         if not base:
             kwds.pop("base")
+        if loffset is not None:
+            if pd.__version__ < "2.0.0":
+                kwds.update({"loffset": loffset})
+            else:
+                warnings.warn(
+                    DeprecationWarning("`loffset` parameter is deprecated in pandas>=2.0.0, ignoring input argument."),
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
```

Comment:
If I am not mistaken, you need to `kwds.pop("loffset")`
```suggestion
            if pd.__version__ >= "2.0.0" and "loffset" in kwds:
                kwds.pop("loffset")
                warnings.warn(
                    DeprecationWarning("`loffset` parameter is deprecated in pandas>=2.0.0, ignoring input argument."),
                    DeprecationWarning,
                    stacklevel=2,
                )
```

Reply:
Also, it is "deprecated" since 1.1 and was removed in 2.0.

Reply:
Oh and the lexical sort of string "2.0.0" does not bode well with double digit minor versions!

Reply:
Ah good call! I can regex for the major version and then compare it as an integer.

Will update the warning phrasing as well, to reflect what you mentioned regarding deprecation v. removable

Reply:
If you really want to do version parsing I'd encourage using [`packaging`'s handling to get an actual version object](https://packaging.pypa.io/en/stable/version.html) rather than trying to roll your own. Version numbers are surprisingly complicated!

Reply:
Here is the stacktrace when i run with pandas>=2.0.0 using the suggestion above (without ignoring the param or sending the warning).. it's not particularly informative, but it does get the point across that `loffset` isn't expected. I think going with this seems reasonable though because it pushes the constraint onto pandas versioning without extra overhead from swifter

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[7], line 1
----> 1 df.swifter.resample(rule=timedelta(days=2), loffset=timedelta(days=2)).apply(lambda x: x**2)

File /mnt/swifter/swifter.py:210, in _SwifterObject.resample(self, rule, axis, closed, label, convention, kind, loffset, base, on, level, origin, offset)
    207 if loffset is not None:
    208     kwds.update({"loffset": loffset})
--> 210 return Resampler(
    211     self._obj,
    212     npartitions=self._npartitions,
    213     dask_threshold=self._dask_threshold,
    214     scheduler=self._scheduler,
    215     progress_bar=self._progress_bar,
    216     progress_bar_desc=self._progress_bar_desc,
    217     allow_dask_on_strings=self._allow_dask_on_strings,
    218     force_parallel=self._force_parallel,
    219     **kwds,
    220 )

File /mnt/swifter/swifter.py:791, in Resampler.__init__(self, pandas_obj, npartitions, dask_threshold, scheduler, progress_bar, progress_bar_desc, allow_dask_on_strings, force_parallel, **kwds)
    789 self._resampler_kwds = kwds.copy()
    790 self._comparison_pd = self._obj_pd.iloc[: self._SAMPLE_SIZE]
--> 791 self._sample_pd = self._sample_pd.resample(**kwds)
    792 self._obj_pd = self._obj_pd.resample(**kwds)
    793 # Setting dask dataframe `self._obj_dd` to None when there are 0 `self._nrows`
    794 # because swifter will immediately return the pandas form during the apply
    795 # function if there are 0 `self._nrows`

TypeError: resample() got an unexpected keyword argument 'loffset'
```